HeterogeneousCore/CUDAUtilities/interface/host_unique_ptr.h

```diff
@@ -10,11 +11,32 @@ namespace cms {
   namespace cuda {
     namespace host {
       namespace impl {
-        // Additional layer of types to distinguish from host::unique_ptr
+
+        enum MemoryType : bool {
```

Comment:
How about `enum class`?

Reply:
I didn't find that using `enum class` would add any benefit here, and it would make the use below slightly more verbose:
```diff
-   host::impl::kPinned};
+   host::impl::MemoryType::kPinned};
```
_etc._

Reply:
On second thought, while it doesn't add much, it's also not very intrusive since the enum is used only within this file.